tests/CMakeLists.txt

```diff
@@ -5,6 +5,9 @@ include_directories(
 
 # all *.cpp files here will be used.
 file(GLOB test_sources *.cpp)
+if(NOT SML_FOUND)
+  list(REMOVE_ITEM test_sources MeterSML.cpp)
```

Comment:
Is this sufficient? I was having a similar problem and needed to use:
list(REMOVE_ITEM test_sources ${CMAKE_CURRENT_SOURCE_DIR}/MeterSML.cpp)